opa-filter-autoconfigure/src/test/groovy/com/bisnode/opa/spring/security/filter/autoconfigure/OpaFilterConfigurerSpec.groovy

```diff
@@ -53,53 +61,70 @@ class OpaFilterConfigurerSpec extends Specification {
 
     def 'should return 403 on opa deny'() {
         given:
-            wireMockServer.stubFor(any(anyUrl())
-                    .willReturn(aResponse()
-                            .withStatus(200)
-                            .withHeader(ContentType.HEADER_NAME, APPLICATION_JSON)
-                            .withBody('{"result": {"allow": false}}')
-                    )
-            )
+          wireMockServer.stubFor(any(anyUrl())
+                  .willReturn(aResponse()
+                          .withStatus(200)
+                          .withHeader(ContentType.HEADER_NAME, APPLICATION_JSON)
+                          .withBody('{"result": {"allow": false}}')
+                  )
+          )
         when:
-            restClient.get(path: '/', headers: ["Authorization": BASIC_AUTH_HEADER])
+          restClient.get(path: '/', headers: ["Authorization": BASIC_AUTH_HEADER])
 
         then:
-            HttpResponseException e = thrown(HttpResponseException)
-            e.response.status == 403
+          HttpResponseException e = thrown(HttpResponseException)
+          e.response.status == 403
+    }
+
+    def 'should publish AuthorizationFailureEvent on opa deny'() {
+        given:
+          wireMockServer.stubFor(any(anyUrl())
+                  .willReturn(aResponse()
+                          .withStatus(200)
+                          .withHeader(ContentType.HEADER_NAME, APPLICATION_JSON)
+                          .withBody('{"result": {"allow": false}}')
```

Comment:
Do we have a test for OPA returning result without allow at all? Or just an empty response?

Reply:
And, should events be published on error responses as well or that's meant to be handled by the usual exception handling only?

Reply:
Events are just additional way to inform developer that access was denied, so they can react to this in any way they want. In our case we are using it to check differences between OPA decision and our legacy solution decision.

Reply:
> Do we have a test for OPA returning result without allow at all? Or just an empty response?

We don't, i will create test that will check that if we will get missing values we will deny.